File: client-core/src/main/java/hivens/core/api/IServerListService.java
package hivens.core.api;

import hivens.core.api.model.ServerProfile;
import java.util.List;
import java.util.concurrent.CompletableFuture;

public interface IServerListService {
    /**
     * Асинхронно получает список профилей серверов.
     */
    CompletableFuture<List<ServerProfile>> fetchProfiles();
}

File: client-core/src/main/java/hivens/core/api/IServerListService.kt
package hivens.core.api

import hivens.core.api.model.ServerProfile
import java.util.concurrent.CompletableFuture

interface IServerListService {
    /**
     * Асинхронно получает список профилей серверов.
     */
    fun fetchProfiles(): CompletableFuture<List<ServerProfile>>
}